### PR TITLE
Bugfix: batch calculation with abc=True

### PR DESCRIPTION
**Pull request by So-Takamoto**
I found that test function `test_calc_energy_force_stress_device_batch_abc` unintentionally ignores `abc` argument.

This PR modified related implementation to work it.

In addition, corner case correspondence when the total number of atom is zero is also added. (`n_graphs` cannot be calculated from `batch_edge` when `len(batch_edge) == 0`.)

### CODE REVIEW

File: tests/test_torch_dftd3_calculator_batch.py [hunk 1]
```diff
@@ -77,35 +77,35 @@ def _test_calc_energy_force_stress(
...
-def test_calc_energy_force_stress_device_batch_abc(damping, old, device, dtype):
-    """Test2-2: check device, dtype dependency. with only various damping method."""
+def test_calc_energy_force_stress_device_batch_abc(damping, old, atoms_list, device, dtype):
+    """Test2-3: check device, dtype dependency. with only various damping method."""
     xc = "pbe"
     abc = True
-    atoms_list = _create_atoms()
     _test_calc_energy_force_stress(
-        damping, xc, old, atoms_list, device=device, dtype=dtype, cnthr=7.0
+        damping, xc, old, atoms_list, device=device, dtype=dtype, abc=abc, cnthr=7.0
```
Comment (So-Takamoto) [inline tests/test_torch_dftd3_calculator_batch.py:108]: Added `abc=abc` here.

File: torch_dftd/functions/dftd3.py [hunk 2]
```diff
@@ -250,7 +250,10 @@ def edisp(
         g = e68.sum()[None]
     else:
         # (n_graphs,)
-        n_graphs = int(batch[-1]) + 1
+        if batch.size()[0] == 0:
+            n_graphs = 1
+        else:
+            n_graphs = int(batch[-1]) + 1
```
Comment (So-Takamoto) [inline torch_dftd/functions/dftd3.py:256]: For zero-sized batch

File: torch_dftd/nn/base_dftd_module.py [hunk 3]
```diff
@@ -133,7 +133,10 @@ def calc_energy_and_forces(
         if batch is None:
             results_list = [{"energy": E_disp.item(), "forces": forces.cpu().numpy()}]
         else:
-            n_graphs = int(batch[-1]) + 1
+            if batch.size()[0] == 0:
+                n_graphs = 1
+            else:
+                n_graphs = int(batch[-1]) + 1
```
Comment (So-Takamoto) [inline torch_dftd/nn/base_dftd_module.py:139]: For zero-sized batch

File: torch_dftd/torch_dftd3_calculator.py [hunk 4]
```diff
@@ -156,14 +156,17 @@ def batch_calculate(self, atoms_list=None, properties=["energy"], system_changes
...
+            cell_batch = torch.zeros((0, 3, 3))
+        else:
+            cell_batch = torch.stack(
+                [
+                    torch.eye(3, device=self.device, dtype=self.dtype)
+                    if d["cell"] is None
+                    else d["cell"]
+                    for d in input_dicts_list
+                ]
+            )
```
Comment (So-Takamoto) [inline torch_dftd/torch_dftd3_calculator.py:169]: For zero-sized batch

File: torch_dftd/functions/dftd3.py [hunk 5]
```diff
@@ -262,29 +265,40 @@ def edisp(
...
         edge_index_abc = edge_index[:, within_cutoff]
+        batch_edge_abc = None if batch_edge is None else batch_edge[within_cutoff]
         # c6_abc = c6[within_cutoff]
         shift_abc = None if shift is None else shift[within_cutoff]
 
         n_atoms = Z.shape[0]
         if not bidirectional:
             # (2, n_edges) -> (2, n_edges * 2)
-            edge_index_abc = torch.cat([edge_index_abc, edge_index_abc.flip(dims=[1])], dim=1)
+            edge_index_abc = torch.cat([edge_index_abc, edge_index_abc.flip(dims=[0])], dim=1)
```
Comment (corochann) [inline torch_dftd/functions/dftd3.py:275]: Is this tested?

Comment (So-Takamoto) [reply torch_dftd/functions/dftd3.py:275]: I have added a test for `bidirectional=True` and `abc=True`.
I confirmed that the tests fail with the previous `dims=[1]` version.

**Review by corochann — APPROVED** [review]
LGTM, thank you for fix!
